lib/interpreters/periodic_visits.rb

```diff
@@ -371,36 +373,63 @@ def associate_skills(new_vehicle, vehicle_day_index)
       end
     end
 
-    def compute_possible_days(vrp)
-      # for each of this service's visits, computes first and last possible day to be assigned
-      vrp.services.each{ |service|
-        day = @schedule_start
-        nb_services_seen = 0
-
-        # first possible day
-        while day <= @schedule_end && nb_services_seen < service.visits_number
-          if service.unavailable_days.include?(day) || vrp.vehicles.none?{ |v| v.available_at(day) }
-            day += 1
+    def possible_days_from_possible_day(service, possible_days, vehicles, lapse, options)
+      day = @schedule_start
+      nb_services_seen = 0
+      set = []
+      while day <= @schedule_end && nb_services_seen < service.visits_number
+        # it is possible that user provided last_performed_visit_date/day_index or filled first_possible_days field
+        # we assume if there are X values in first_possible_days then it corresponds to X first visits of this service
+        # user needs to provide nil value if he wants to skip one service when providing first_possible_days
+        day =
+          if lapse
+            [possible_days[nb_services_seen], day].compact.max
```

Comment:
```suggestion
            [possible_days[visit_index], day].compact.max
```

Reply:
I assume possible_days is an array ? How could you compare and array with a day ?

Reply:
possible_days is an array of integers, hence possible_days[nb_services_seen] is an integer. We are comparing day with possible_days[nb_services_seen] :)